middlewares/authenticate.js

```diff
@@ -1,6 +1,17 @@
 const authService = require('../services/authService')
 const users = require('../models/users')
 
+/**
+ * Middleware to check if access has been restricted for this user.
+ */
+const checkLimitedAccess = async (req, res, next) => {
+  const { restricted } = req.userData
```

Comment:
Is this `restricted` field going to be on the root level of the user data itself or inside an object like `status`? 🤔